### PR TITLE
feat: replaced packages @edx/paragon & @edx/frontend-build to use openedx namespace

### PR DESCRIPTION
**Pull request by BilalQamar95**
**Description**
- Replaced packages `@edx/paragon` & `@edx/frontend-build` to use `openedx` namespace
- Major version upgrades, Updating `frontend-build` to `v14` & `frontend-platform` to `v8` along with respective edx packages
- Updated `ts-jest` to `v29` along with respective packages
- Fixed failing tests post upgrade by making necessary upgrades

### CODE REVIEW

File: src/components/learner-credit-management/tests/BudgetDetailPage.test.jsx
Comment (adamstankiewicz) [inline src/components/learner-credit-management/tests/BudgetDetailPage.test.jsx:933]: `userEvent.click`  should not be used within an `act`. See https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#wrapping-things-in-act-unnecessarily for more details.

File: package.json
Comment (abdullahwaheed) [reply package.json:116]: i don't this we should add alpha as peer dependency since alpha is supposed to be a pre-release. We should keep peer dependencies for actual releases

File: src/components/learner-credit-management/tests/BudgetDetailPage.test.jsx
Comment (BilalQamar95) [reply src/components/learner-credit-management/tests/BudgetDetailPage.test.jsx:933]: Since tests were failing without it `act` was used initially to ensure that all updates to the component (like state changes and DOM updates) were flushed and applied before assertions without these test was failing. However you are right, I have replaced it with `waitFor` which automatically handles the flushing of updates within its callback, waiting for the expected condition to be true which seems to have resolved the issue.

File: src/components/settings/SettingsLMSTab/LMSSelectorPage.tsx
Comment (adamstankiewicz) [inline src/components/settings/SettingsLMSTab/LMSSelectorPage.tsx:41]: The `<p>` above with the text "Select the LMS or LXP you want to integrate..." is a more useful label for the `SelectableBox.Set` than "Let's get started"... 

"Let's get started" doesn't really describe the choices in the `SelectableBox.Set`.

File: package.json
Comment (BilalQamar95) [reply package.json:94]: This was intentional as v14 contains `jest` v29 upgrade & this PR focuses on `@edx/paragon` & `@edx/frontend-build` to `openedx` namespace switch. There is another [PR (currently drafted)](https://github.com/openedx/frontend-app-admin-portal/pull/1202/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R94) for `jest` v29 upgrade, containing upgrade to `frontend-build` v14 along with respective `edx` packages, which will be updated and ready for review once this PR is merged

Comment (adamstankiewicz) [reply package.json:94]: I'm not really following why this work needs to be split into 2 separate PRs at this point. Why can't the upgrades from the [draft PR](https://github.com/openedx/frontend-app-admin-portal/pull/1202/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R94) just be migrated to this PR since they are so related anyways? IMHO, I would have considered Jest v29 upgrade (and its other Jest-related package upgrades) as part of the original scope of migrating to latest version of `@openedx/frontend-build` in this PR directly; personally don't really see why this work needs to be split across 2 PRs.

Either way, I'm not going to push back further if you feel the incremental upgrade and separate upgrade is truly necessary versus bringing the other upgrades into this PR (given Jest upgrade, etc. actually _is_ related to the scope of migrating to `@openedx/frontend-build`, IMHO).

File: src/components/settings/SettingsLMSTab/LMSSelectorPage.tsx
Comment (adamstankiewicz) [reply src/components/settings/SettingsLMSTab/LMSSelectorPage.tsx:41]: @BilalQamar95 Bump on this comment. Can we move the `id="lms-selection-heading"` to the `<p>` instead so it's a more useful/semantically correct message?

```
<p id="lms-selection-heading">Select the LMS or LXP you want to integrate with edX For Business.</p>
```

File: package.json
Comment (BilalQamar95) [reply package.json]: Thank you for your feedback. My rationale behind splitting the work into 2 different PRs was based on 
1. Scope and Complexity Management: The primary focus of the current PR is the namespace switch to `@openedx` for `@edx/paragon` and `@edx/frontend-build`. This was scoped as a standalone epic to ensure a clean transition without introducing additional variables. Also the `frontend-build` `v14` brings in several major package upgrades further adding to the complexity which when handled separately could ensure a more focused and manageable review process. 
2. Project Timeline and Dependencies: When the PR was initially created, `frontend-build `v14 was not yet available. It was intended to address the namespace switch only. The subsequent release of `frontend-build `v14 necessitated additional work, which was scoped into a separate Jest v29 epic
3. Given the significant number of file changes and potential issues arising from multiple major upgrades, splitting the work allows for easier tracking and isolation of issues, also incremental approach reduces the risk of introducing widespread breaking changes and simplifies troubleshooting if things go south.
4. The upgrades correspond to different epics, one focused on the namespace switch and the other on upgrading Jest and related packages. Keeping them separate maintains alignment with the objectives.

I understand the perspective that combining the upgrades might seem more efficient.  I have decided to combine the two considering the review process duration and the fact that the change itself, while widespread, is not overly complicated, it will ultimately expedite the merge process and reduce the overall time required for reviews by having single review for both features.

File: package.json
Comment (adamstankiewicz) [inline package.json:40]: nit: pin dependency

**Review by adamstankiewicz — APPROVED** [review]
LGTM (with tiny nit around pinning the Paragon dependency). Thanks for addressing the feedback, @BilalQamar95. Happy to help QA/verify on stage once merged.

**Comment by BilalQamar95** [comment]
> LGTM (with tiny nit around pinning the Paragon dependency). Thanks for addressing the feedback, @BilalQamar95. Happy to help QA/verify on stage once merged.

Thank you for the approval & for QA/verification on stage. I have pinned the paragon and it will be merged once CI clears

**Pull request merged**
